esp/services/ws_logaccess/WsLogAccessService.cpp

```diff
@@ -52,63 +52,145 @@ LogAccessTimeRange requestedRangeToLARange(IConstTimeRange & reqrange)
 
-            logFetchOptions.setFilter(getAudienceLogAccessFilter(targetAud));
-            break;
+            return getAudienceLogAccessFilter(targetAud);
         }
         case CLogAccessType_BySourceInstance:
         {
-            logFetchOptions.setFilter(getInstanceLogAccessFilter(searchByValue));
-            break;
+            return getInstanceLogAccessFilter(searchByValue);
         }
         case CLogAccessType_BySourceNode:
         {
-            logFetchOptions.setFilter(getHostLogAccessFilter(searchByValue));
+            return getHostLogAccessFilter(searchByValue);
             break;
         }
+        case CLogAccessType_ByFieldName:
+        {
+            return getColumnLogAccessFilter(serchField, searchByValue);
+        }
         case LogAccessType_Undefined:
         default:
             throw makeStringException(-1, "Invalid remote log access request type");
     }
+}
+
+ILogAccessFilter * processLogFilter(IConstLogFilter * logFilter)
+{
+    if (logFilter)
+        return processLogFilterByFields(logFilter->getLogCategory(), logFilter->getSearchByValue(), logFilter->getSearchField());
+    else
+        return nullptr;
+}
+
+ILogAccessFilter * processBinaryLogFilter(IConstBinaryLogFilter * binaryfilter)
+{
+    if (!binaryfilter)
+        return nullptr;
+    
+    ILogAccessFilter * leftFilter = nullptr;
+    if (binaryfilter->getLeftBinaryFilter().ordinality() == 0)
+    {
+        leftFilter = processLogFilter(&binaryfilter->getLeftFilter());
+    }
+    else
+    {
+        leftFilter = processBinaryLogFilter(&binaryfilter->getLeftBinaryFilter().item(0));
+        if (binaryfilter->getLeftBinaryFilter().ordinality() > 1)
+        {
+            WARNLOG("TOO MANY Left FILTERS ENCOUNTERED");
+        }
+    }
+
+    if (!leftFilter)
+        throw makeStringExceptionV(-1, "WsLogAccess: Empty left filter encountered");
+
+    switch (binaryfilter->getOperator())
+    {
+    case CLogAccessFilterOperator_NONE:
+    case LogAccessFilterOperator_Undefined: //no operator found
+        //if (rightFilter != nullptr)
+        //     WARNLOG("right FILTER ENCOUNTERED but no valid operator");
+        return leftFilter;
+    case CLogAccessFilterOperator_AND:
+    case CLogAccessFilterOperator_OR:
+    {
+        ILogAccessFilter * rightFilter = nullptr;
+        if (binaryfilter->getRightBinaryFilter().ordinality() == 0)
+        {
+            rightFilter = processLogFilter(&binaryfilter->getRightFilter());
+        }
+        else
+        {
+            rightFilter = processBinaryLogFilter(&binaryfilter->getRightBinaryFilter().item(0));
+            if (binaryfilter->getRightBinaryFilter().ordinality() > 1)
+            {
+                WARNLOG("TOO MANY right FILTERS ENCOUNTERED");
+            }
+        }
+        return getBinaryLogAccessFilterOwn(leftFilter, rightFilter, cLogAccessFilterOperator2LogAccessFilterType(binaryfilter->getOperator()));
```

Comment:
I would have named this function something else besides "get".  More like "createBinaryLogAccessFilterOwn()".  But I see there are quite a few of these functions named similarly.

Reply:
will leave pre-existing function name as is for now.
I'm looking into an automated pre-code review scan for all PRs where we can enforce a lot of our unwritten styles/protocols (not sure if this one could be articulated).

Reply:
@GordonSmith / @jeclrsg any comments before we finalize this?